src/goto-instrument/contracts/dynamic-frames/dfcc_infer_loop_assigns.cpp

```diff
@@ -46,18 +47,91 @@ depends_on(const exprt &expr, std::unordered_set<irep_idt> identifiers)
   return false;
 }
 
+/// Collect identifiers that are local to this loop only
+/// (excluding nested loop).
```

Comment:
When generating loop information, we want to collect locals that are only local for the given loop_id but not its nested inner loops to avoid instrument a loop instruction more than once.
I remove the condition of excluding nested inner loops from this method. And erase inner loops' locals in `gen_dfcc_loop_info` instead.